File: leetcode/practice/1300-sum-of-mutated-array-closest-to-target/optimized.py
class Solution:
    def findBestValue(self, arr: List[int], target: int) -> int:
        arr.sort(reverse = 1)
        
        tempAns = arr[0]
        while(arr and target > arr[-1]*len(arr)) :
            target -= arr.pop()
        
        if not arr :
            return tempAns
        
        return int(round((target - 0.0001)/len(arr)))

File: leetcode/practice/1300-sum-of-mutated-array-closest-to-target/optimized.cpp
class Solution {
public:
    int findBestValue(vector<int>& arr, int target) ;
};

int Solution :: findBestValue(vector<int>& arr , int target)
{
    sort(begin(arr) , end(arr)) ;
    long value = INT_MAX ;
    int ind = 0 , len = arr.size() ;
    while(ind < len and target > arr[ind]*(len - ind))
    {
        target -= arr[ind++] ;
    }
    
    if(ind == len)
        return arr[ind - 1] ;
    int val = round((target - 0.0001)/(len - ind)) ;
    return val ;
}
